spring-web/src/main/java/org/springframework/web/server/ServerWebExchange.java

```diff
@@ -118,4 +122,19 @@
 	 */
 	boolean checkNotModified(String etag, Instant lastModified);
 
+	/**
+	 * Return a stream of parts parsed as they are received.
+	 * @return a {@link Flux} containing the various parts if the current request is a multipart
+	 * one or one emitting a {@link MultipartException} if the request is not multipart, or if
+	 * implementation-specific problems are encountered (such as exceeding file size limits).
+	 */
+	Flux<Part> getParts();
+
+	/**
+	 * Lazily parse all parts of a multipart request.
+	 * @return a {@link Mono} with a map containing all the parts parsed, with the name as key,
+	 * or a {@link Mono} emitting an error the current request is not a multipart one.
+	 */
+	Mono<MultiValueMap<String, Part>> getAllParts();
+
```

Comment:
I changed the order and also proposed new names in order to be more consistent with `getFormData()`: `Mono<MultiValueMap<String, Part>> getFormParts()` and `Flux<Part> getFormPartsAsFlux()`. Are you ok with these new names?